TRENZ.Lib.RazorMail.System.Net/Extensions/SmtpClientExtensions.cs

```diff
@@ -1,40 +1,40 @@
-﻿using System;
-using System.Net.Mail;
-using System.Threading.Tasks;
-
-namespace TRENZ.Lib.RazorMail.SystemNetExtensions;
-
-public static class SmtpClientExtensions
-{
-    public static Task SendAsync(this SmtpClient client, MailMessage message)
-    {
-        var tcs = new TaskCompletionSource<object?>();
-        Guid sendGuid = Guid.NewGuid();
-
-        SendCompletedEventHandler? handler = null;
-        
-        handler = (_, ea) =>
-        {
-            if (ea.UserState is not Guid userStateGuid || userStateGuid != sendGuid) return;
-            
-            client.SendCompleted -= handler;
-            
-            if (ea.Cancelled)
-            {
-                tcs.SetCanceled();
-            }
-            else if (ea.Error != null)
-            {
-                tcs.SetException(ea.Error);
-            }
-            else
-            {
-                tcs.SetResult(null);
-            }
-        };
-
-        client.SendCompleted += handler;
-        client.SendAsync(message, sendGuid);
-        return tcs.Task;
-    }
+﻿using System;
+using System.Net.Mail;
+using System.Threading.Tasks;
+
+namespace TRENZ.Lib.RazorMail.Extensions;
+
+public static class SmtpClientExtensions
+{
+    public static Task SendAsync(this SmtpClient client, MailMessage message)
```

Comment:
add a `CancellationToken` with a TODO here, maybe

Reply:
I implemented it right away. The token calls `SendAsyncCancel` on the client and if I understand it correctly, the TCS will be marked as cancelled by the `SendCompleted` handler.